app/models.py

```diff
@@ -399,7 +404,13 @@ class User(db.Model):
     password_changed_at = db.Column(db.DateTime, index=False, unique=False,
                                     nullable=False)
     logged_in_at = db.Column(db.DateTime, nullable=True)
+
+    # used to determine whether account is `locked`. field is reset upon successful login or can
+    # be reset manually to "unlock" an account.
     failed_login_count = db.Column(db.Integer, nullable=False, default=0)
+
+    # used by frontends to determine whether view access should be allowed, also
+    # used to police the validity of some operations by the api internally.
```

Comment:
_Actually_, forget that - the place where it does that is in the `User` object serialization itself, which makes perfect sense. Can probably get away with removal of this sencence.